web-marketplace/src/pages/BuyCredits/BuyCredits.tsx

```diff
@@ -80,6 +86,16 @@ export const BuyCredits = () => {
     if (confirmationTokenId) summarizePayment(confirmationTokenId);
   }, [confirmationTokenId, summarizePayment]);
 
+  const noProjectFound =
+    !loadingProjectByOnChainId &&
+    !projectByOnChainId &&
+    !loadingProjectBySlug &&
+    !projectBySlug?.data?.projectBySlug &&
+    !loadingOffchainProjectById &&
+    !offchainProjectByIdData;
```

Comment:
should we use the exact same condition as in ProjectDetails and try to refactor this a bit to have `noProjectFound` returned by `useGetProject`